kratos/processes/apply_ray_casting_interface_recognition_process.h

```diff
@@ -0,0 +1,191 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Daniel Diez
+//
+
+#pragma once
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "processes/apply_ray_casting_process.h"
+
+namespace Kratos
+{
+///@addtogroup Kratos Core
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Applies ray casting to distinguish the color (like in/out) of each node in modelpart
+/** This class is used to define the which nodes are inside, outside or on the interface
+ *  of certain volume described by its contour
+ */
+template<std::size_t TDim = 3>
```

Comment:
```suggestion
template<std::size_t TDim>
```
To make sure the instance we're using (at the end of the day you need to put the `<>` in case you want to take the default value so there is no real advantage at all...).

Reply:
agree